src/DynamoCoreWpf/ViewModels/Watch3D/HelixWatch3DViewModel.cs

```diff
@@ -1978,16 +1986,36 @@ public int Compare(Model3D x, Model3D y)
             var transB = (bool) b.GetValue(AttachedProperties.HasTransparencyProperty);
             result = transA.CompareTo(transB);
 
-            if (result != 0 || !transA) return result;
+            // if only one of transA and transB has transparency, sort by having this property
+            if (result != 0) return result;
 
-            // compare distance
-            var boundsA = a.Bounds;
-            var boundsB = b.Bounds;
-            var cpA = (boundsA.Maximum + boundsA.Minimum)/2;
-            var cpB = (boundsB.Maximum + boundsB.Minimum)/2;
-            var dA = Vector3.DistanceSquared(cpA, cameraPosition);
-            var dB = Vector3.DistanceSquared(cpB, cameraPosition);
-            return -dA.CompareTo(dB);
+            // if both items has transparency, sort by distance
+            if (transA)
+            {
+                // compare distance
+                var boundsA = a.Bounds;
+                var boundsB = b.Bounds;
+                var cpA = (boundsA.Maximum + boundsA.Minimum) / 2;
+                var cpB = (boundsB.Maximum + boundsB.Minimum) / 2;
+                var dA = Vector3.DistanceSquared(cpA, cameraPosition);
+                var dB = Vector3.DistanceSquared(cpB, cameraPosition);
+                result = -dA.CompareTo(dB);
+                return result;
+            }
+
+            // if both items does not have transparency, sort following next order: mesh, line, point
+            var pointA = a is PointGeometryModel3D;
+            var pointB = b is PointGeometryModel3D;
+            result = pointA.CompareTo(pointB);
+
+            if (pointA || pointB)
+            {
+                return result;
+            }
+
+            var lineA = a is LineGeometryModel3D;
+            var lineB = b is LineGeometryModel3D;
+            return lineA.CompareTo(lineB);
```

Comment:
if we get to this point and the object is not a line then won't this throw an exception? lets guard against that.